File: src/entities/Post.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entities;

import java.sql.Timestamp;

/**
 *
 * @author anaso
 */


enum Module {
    MATHEMATIQUE_DE_BASE_1, MATHEMATIQUE_DE_BASE_2, IP_ESSENTIALS, TECHNIQUE_DESTIMATION_POUR_LINGENIEUR
}

public class Post {

    private int id,ownerId,solution_id;
    private String problem;
    private Timestamp timestamp;
    private String status;
    private String module;
    private String imageName;

    public Post() {
        if (status==null)
            status="OPEN";
    }

    public Post(String problem, String module) {
        this.problem = problem;
        this.module = module;
    }

    public Post(int id, String problem, String module, Timestamp timestamp) {
        this.id = id;
        this.problem = problem;
        this.module = module;
        this.timestamp = timestamp;
    }

    public Post(int id, String problem, String imageName) {
        this.id = id;
        this.problem = problem;
        this.imageName = imageName;
    }

    public Post(String problem, String module, String imageName) {
        this.problem = problem;
        this.module = module;
        this.imageName = imageName;
    }

    
    
    public void setImageName(String imageName) {
        this.imageName = imageName;
    }

    public String getImageName() {
        return imageName;
    }

    
    public int getId() {
        return id;
    }

    public String getProblem() {
        return problem;
    }

    public Timestamp getTimestamp() {
        return timestamp;
    }



    public String getModule() {
        return module;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setProblem(String problem) {
        this.problem = problem;
    }

    public void setTimestamp(Timestamp timestamp) {
        this.timestamp = timestamp;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    

    public void setModule(String module) {
        this.module = module;
    }

    public int getOwnerId() {
        return ownerId;
    }

    public int getSolution_id() {
        return solution_id;
    }

    public void setOwnerId(int ownerId) {
        this.ownerId = ownerId;
    }

    public void setSolution_id(int solution_id) {
        this.solution_id = solution_id;
    }

    @Override
    public String toString() {
        return "Post{" + "id=" + id + ", ownerId=" + ownerId + ", solution_id=" + solution_id + ", problem=" + problem + ", timestamp=" + timestamp + ", status=" + status + ", module=" + module + ", imageName=" + imageName + '}';
    }


    
    
    
    

}

File: Symfony4/src/Entity/Post.php
<?php

namespace App\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Vangrg\ProfanityBundle\Validator\Constraints\ProfanityCheck;

/**
 * Post
 *
 * @ORM\Table(name="post", indexes={@ORM\Index(name="fk_solutionId", columns={"solution_id"}), @ORM\Index(name="fk_ownerId", columns={"owner_id"})})
 * @ORM\Entity(repositoryClass="App\Repository\PostRepository")
 */
class Post
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="timestamp", type="datetime", nullable=false, options={"default"="CURRENT_TIMESTAMP"})
     */
    private $timestamp;

    /**
     * @var string
     *
     * @ORM\Column(name="status", type="string", length=12, nullable=false, options={"default"="OPEN"})
     */
    private $status = 'OPEN';

    /**
     * @var string
     *
     * @ORM\Column(name="module", type="string", length=255, nullable=false)
     */
    private $module;

    /**
     * @var string
     *
     * @ORM\Column(name="problem", type="string", length=255, nullable=false)
     * @Assert\NotBlank(message="How can you publish a post with no problem ? ")
     * @ProfanityCheck()
     */
    private $problem;

    /**
     * @var string
     *
     * @ORM\Column(name="image_name", type="string", length=255, nullable=true)
     */
    private $imageName;

    /**
     * @ORM\OneToMany(targetEntity=Comment::class, mappedBy="post", cascade="all")
     */
    private $comments;

    public function __construct()
    {
        $this->comments = new ArrayCollection();
    }

    /**
     * @var \Users
     *
     * @ORM\ManyToOne(targetEntity="Users")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="owner_id", referencedColumnName="Id")
     * })
     */
    private $owner;


    /**
     * @var \Comment
     *
     * @ORM\ManyToOne(targetEntity="Comment")
     * @ORM\OrderBy({ "timestamp" = "DESC" })
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="solution_id", referencedColumnName="id")
     * })
     */
    private $solution;


    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTimestamp(): ?\DateTimeInterface
    {
        return $this->timestamp;
    }

    public function setTimestamp(\DateTimeInterface $timestamp): self
    {
        $this->timestamp = $timestamp;

        return $this;
    }

    public function getStatus(): ?string
    {
        return $this->status;
    }

    public function setStatus(string $status): self
    {
        $this->status = $status;

        return $this;
    }

    public function getModule(): ?string
    {
        return $this->module;
    }

    public function setModule(string $module): self
    {
        $this->module = $module;

        return $this;
    }

    public function getProblem(): ?string
    {
        return $this->problem;
    }

    public function setProblem(string $problem): self
    {
        $this->problem = $problem;

        return $this;
    }

    public function getImageName()
    {
        return $this->imageName;
    }

    public function setImageName($imageName): self
    {
        $this->imageName = $imageName;

        return $this;
    }

//    public function getImagePath()
//    {
//        return 'images/'.$this->getImageFilename();
//    }


    public function getOwner(): ?Users
    {
        return $this->owner;
    }

    public function setOwner(?Users $owner): self
    {
        $this->owner = $owner;

        return $this;
    }

    public function getSolution(): ?Comment
    {
        return $this->solution;
    }

    public function setSolution(?Comment $solution): self
    {
        $this->solution = $solution;

        return $this;
    }

    /**
     * @return Collection|Comment[]
     */
    public function getComments(): Collection
    {
        return $this->comments;
    }

    public function setComment(Comment $comment): self
    {
        if (!$this->comments->contains($comment)) {
            $this->comments[] = $comment;
            $comment->setPost($this);
        }

        return $this;
    }

    public function removeComment(Comment $comment): self
    {
        if ($this->comments->removeElement($comment)) {
            // set the owning side to null (unless already changed)
            if ($comment->getPost() === $this) {
                $comment->setPost(null);
            }
        }

        return $this;
    }

    public function __toString()
    {
        return $this->problem;
    }


}
